### PR TITLE
Bumped `starknet-rs`

### PR DESCRIPTION
**Pull request by integraledelebesgue**
Closes [#2352](https://github.com/orgs/foundry-rs/projects/3?pane=issue&itemId=73510958)

## Introduced changes
- `starknet-rs` and `starknet-crypto` dependencies have been updated to the latest commit [660a732](https://github.com/xJonathanLEI/starknet-rs/commit/660a7323169bbcdaea71b4f0e939631e3aaabd45)
- [test](https://github.com/foundry-rs/starknet-foundry/blob/b16628953cb318d835f054968c2df3f50a63ef22/crates/sncast/tests/e2e/declare.rs#L233) related to CASM artifact edge case is no longer ignored
- A new `Felt` type replaced `FieldElement`
- Appropriate changes in `conversions` crate 

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/conversions/src/felt252.rs [hunk 1]
```diff
@@ -4,13 +4,13 @@ use crate::{
...
 };
-use starknet::core::types::FieldElement;
+use starknet::core::types::Felt;
 use starknet_api::core::{ClassHash, ContractAddress, EntryPointSelector, Nonce};
 use starknet_types_core::felt::{Felt as Felt252, FromStrError};
 use std::vec;
 
-impl FromConv<FieldElement> for Felt252 {
-    fn from_(value: FieldElement) -> Felt252 {
+impl FromConv<Felt> for Felt252 {
```
Comment (ksew1) [inline crates/conversions/src/felt252.rs:12]: It is the same type you can just do:

```rs
impl FromConv<Felt252> for Felt252 {
    fn from_(value: Felt252) -> Felt252 {
        value
    }
}
```

Comment (ksew1) [reply crates/conversions/src/felt252.rs:12]: Or leave what you deleted:

```rs
impl<T> FromConv<T> for T {
    fn from_(value: T) -> Self {
        value
    }
}
```

File: crates/sncast/src/response/structs.rs [hunk 2]
```diff
@@ -22,22 +17,12 @@ impl Serialize for Decimal {
...
-    }
-}
-
 fn serialize_as_decimal<S>(value: &Felt, serializer: S) -> Result<S::Ok, S::Error>
 where
     S: Serializer,
 {
-    let val = value.0;
-    serializer.serialize_str(&format!("{val:#}"))
+    // let val = value.0;
```
Comment (ksew1) [inline crates/sncast/src/response/structs.rs:24]:
```suggestion
```

File: crates/sncast/src/starknet_commands/call.rs [hunk 3]
```diff
@@ -45,10 +45,8 @@ pub async fn call(
         entry_point_selector,
         calldata,
     };
-    let res = provider
-        .call(function_call, block_id)
-        .await
-        .map(|v| v.into_iter().map(Felt).collect());
+    let res = provider.call(function_call, block_id).await;
+    // .map(|v| v.into_iter().map(Felt).collect());
```
Comment (ksew1) [inline crates/sncast/src/starknet_commands/call.rs:49]:
```suggestion
```

File: crates/snforge-scarb-plugin/src/parse.rs [hunk 4]
```diff
@@ -31,7 +32,11 @@ pub fn parse<T: AttributeInfo>(
         .items(db)
         .elements(db);
 
-    if let Some(ModuleItem::FreeFunction(func)) = elements.into_iter().next() {
+    if let Some(ModuleItem::FreeFunction(func)) = elements
+        .into_iter()
+        .skip_while(|element| matches!(element, ModuleItem::HeaderDoc(_)))
```
Comment (ksew1) [inline crates/snforge-scarb-plugin/src/parse.rs:37]: Why this change?

File: crates/sncast/tests/e2e/declare.rs [hunk 5]
```diff
@@ -228,8 +228,9 @@ async fn test_invalid_version_and_token_combination(fee_token: &str, version: &s
         format!("Error: {fee_token} fee token is not supported for {version} declaration."),
     );
 }
+
 #[tokio::test]
-#[ignore = "Expand the contract's code to more complex or wait for fix: https://github.com/xJonathanLEI/starknet-rs/issues/649#issue-2469861847"]
+#[ignore = "#2459"]
```
[the remark this answers is not in the document]
Comment (integraledelebesgue) [reply crates/sncast/tests/e2e/declare.rs:233]: Sadly the problem remains on Devnet - see [#2459](https://github.com/orgs/foundry-rs/projects/3/views/2?pane=issue&itemId=79387057)

**Review by ddoktorski — APPROVED** [review]
